scripts/github-translation-status.mjs

```diff
@@ -0,0 +1,462 @@
+		pageData.lastChange = latest(oldPageData.lastChange, gitHistory.lastCommitDate);
+		pageData.lastCommitMsg = gitHistory.lastCommitMessage;
+		pageData.lastMajorChange = latest(oldPageData.lastMajorChange, gitHistory.lastMajorCommitDate);
+		pageData.lastMajorCommitMsg = gitHistory.lastMajorCommitMessage;
+		
+		return pageData;
+	}
+
+	tryGetFrontMatterBlock (filePath) {
+		const contents = fs.readFileSync(filePath, 'utf8');
+		const matches = contents.match(/^\s*---([\S\s]*?)\n---/);
+		if (!matches)
+			return '';
+		return matches[1];
+	}
+
+	async getGitHistory (filePath) {
+		const gitLog = await this.git.log({
+			file: filePath,
+			strictDate: true,
+		});
+
+		const lastCommit = gitLog.latest;
+		
+		// Attempt to find the last "major" commit, ignoring any commits that
+		// usually do not require translations to be updated
+		const lastMajorCommit = gitLog.all.find(logEntry => {
+			return !logEntry.message.match(/(minor|typo|broken link|i18nReady)/i);
+		}) || lastCommit;
+
+		return {
+			lastCommitMessage: lastCommit.message,
+			lastCommitDate: this.toUtcString(lastCommit.date),
+			lastMajorCommitMessage: lastMajorCommit.message,
+			lastMajorCommitDate: this.toUtcString(lastMajorCommit.date),
+		};
+	}
+
+	toUtcString (date) {
+		return new Date(date).toISOString();
+	}
+
+	renderTranslationStatusByContent ({ pages }) {
+		const arrContent = this.getTranslationStatusByContent({ pages });
+		const lines = [];
+
+		lines.push(`| Content | ${this.targetLanguages.join(' | ')} |`);
+		lines.push(`| :--- | ${this.targetLanguages.map(_ => ':---:').join(' | ')} |`);
+
+		arrContent.forEach(content => {
+			const cols = [];
+			cols.push(`[${content.subpath}](${content.githubUrl})`);
+			cols.push(...this.targetLanguages.map(lang => {
+				const translation = content.translations[lang];
+				if (translation.isMissing)
+					return '<span title="Missing">❌</span>';
+				if (translation.isOutdated)
+					return '<span title="Needs updating">🔄</span>';
+				return '<span title="Completed">✔</span>';
+			}));
+			lines.push(`| ${cols.join(' | ')} |`);
+		});
+
+		lines.push(`\n<sup>❌ Missing &nbsp; 🔄 Needs updating &nbsp; ✔ Completed</sup>`);
+
+		return lines.join('\n');
+	}
+
+	renderTranslationTodosByLanguage ({ pages }) {
+		const arrContent = this.getTranslationStatusByContent({ pages });
+		const lines = [];
+
+		this.targetLanguages.forEach(lang => {
+			const missing = arrContent.filter(content => content.translations[lang].isMissing);
+			const outdated = arrContent.filter(content => content.translations[lang].isOutdated);
+			lines.push('<details>');
+			lines.push(
+				`<summary><b>` +
+				`${lang}: ` +
+				`${missing.length} missing, ${outdated.length} needs updating` +
+				`</b></summary>`
+			);
+			lines.push(``);
+			if (missing.length > 0) {
+				lines.push(`##### Missing`);
+				lines.push(...missing.map(content =>
+					`- [${content.subpath}](${content.githubUrl})`
```

Comment:
Bonus: would it be helpful to offer a GitHub create link here?

For example:

`https://github.com/withastro/docs/new/main/src/pages/{LANG}?filename={PAGE/SLUG.md}`

The `filename` param seems to support `/` to create subdirectories, so you just need the language and then the full slug including directories if present can be passed to `filename`.

It’s even possible to pre-fill the body:

```js
const createUrl = new URL(`https://github.com/withastro/docs/new/main/src/pages/${lang}`);
createUrl.searchParams.set('filename', content.subpath);
createUrl.searchParams.set('value', `---
layout: ~/layouts/MainLayout.astro
title:
description:
---
`);
```

Reply:
This is an amazing idea!